packages/blade/src/components/Box/BaseBox/types.ts

```diff
@@ -1,44 +1,155 @@
+  padding: SpacingValueType | ArrayOfMaxLength4<SpacingValueType>;
+  paddingX: SpacingValueType;
+  paddingY: SpacingValueType;
+  paddingTop: SpacingValueType;
+  paddingRight: SpacingValueType;
+  paddingBottom: SpacingValueType;
+  paddingLeft: SpacingValueType;
+}>;
+
+type MarginProps = MakeObjectResponsive<{
+  /**
+   * Margin shorthand
+   *
+   * For token to pixel conversion, checkout -
+   *
+   * {@linkcode https://blade.razorpay.com/?path=/story/tokens-spacing--page&globals=measureEnabled:false Spacing Token Ref}
+   */
+  margin: SpacingValueType | ArrayOfMaxLength4<SpacingValueType>;
+  marginX: SpacingValueType;
+  marginY: SpacingValueType;
+  marginTop: SpacingValueType;
+  marginRight: SpacingValueType;
+  marginBottom: SpacingValueType;
+  marginLeft: SpacingValueType;
+}>;
+
+type LayoutProps = MakeObjectResponsive<
+  {
+    height: SpacingValueType;
+    minHeight: SpacingValueType;
+    maxHeight: SpacingValueType;
+    width: SpacingValueType;
+    minWidth: SpacingValueType;
+    maxWidth: SpacingValueType;
+  } & Pick<CSSObject, 'display' | 'overflow'>
+>;
+
+type FlexboxProps = MakeObjectResponsive<
+  {
+    gap: SpacingValueType;
+    rowGap: SpacingValueType;
+    columnGap: SpacingValueType;
+  } & Pick<
+    CSSObject,
+    | 'flex'
+    | 'flexWrap'
+    | 'flexDirection'
+    | 'flexGrow'
+    | 'flexShrink'
+    | 'flexBasis'
+    | 'alignItems'
+    | 'alignContent'
+    | 'alignSelf'
+    | 'justifyItems'
+    | 'justifyContent'
+    | 'justifySelf'
+    | 'order'
+  >
+>;
+
+type PositionProps = MakeObjectResponsive<
+  {
+    top: SpacingValueType;
+    right: SpacingValueType;
+    bottom: SpacingValueType;
+    left: SpacingValueType;
+  } & Pick<CSSObject, 'position' | 'zIndex'>
+>;
+
+type GridProps = MakeObjectResponsive<
+  Pick<
+    CSSObject,
+    | 'grid'
+    | 'gridColumn'
+    | 'gridRow'
+    | 'gridRowStart'
+    | 'gridRowEnd'
+    | 'gridArea'
+    | 'gridAutoFlow'
+    | 'gridAutoRows'
+    | 'gridAutoColumns'
+    | 'gridTemplate'
+    | 'gridTemplateAreas'
+    | 'gridTemplateColumns'
+    | 'gridTemplateRows'
+  >
+>;
+
+type ColorObjects = 'feedback' | 'surface' | 'action';
+type BackgroundColorString<T extends ColorObjects> = `${T}.background.${DotNotationColorStringToken<
+  Theme['colors'][T]['background']
+>}`;
+
+type BackgroundColorType =
+  | BackgroundColorString<'feedback'>
+  | BackgroundColorString<'surface'>
+  | BackgroundColorString<'action'>
+  | (string & Record<never, never>);
+
+type VisualProps = MakeObjectResponsive<
+  {
+    borderRadius: keyof Border['radius'];
+    backgroundColor: BackgroundColorType;
+    background: BackgroundColorType;
+  } & Pick<CSSObject, 'transform'>
+>;
+
+type BaseBoxProps = Partial<
+  PaddingProps &
+    MarginProps &
+    LayoutProps &
+    FlexboxProps &
+    PositionProps &
+    GridProps &
+    VisualProps & {
+      children: React.ReactNode | React.ReactNode[];
+      className?: string;
+      id?: string;
+    }
+>;
+
```

Comment:
Will have to add that setup. We can explore this later because `BaseBox` is something that is used internally in our codebase as well. So if some TS fails in future, we'll likely get to know it internally.